python-sdk/nuscenes/predict/tests/run_covernet.py

```diff
@@ -0,0 +1,96 @@
+# nuScenes dev-kit.
+# Code written by Freddy Boulton, 2020.
+
+"""
+Regression test to see if CoverNet implementation can overfit on a single example.
+"""
+
+import argparse
+
+import math
+import numpy as np
+import torch
+import torch.optim as optim
+from torch.utils.data import DataLoader, IterableDataset
+
+from nuscenes.predict.models.backbone import MobileNetBackbone
+from nuscenes.predict.models.covernet import CoverNet, ConstantLatticeLoss
+
+
+def generate_trajectory(theta: float) -> torch.Tensor:
+    trajectory = torch.zeros(6, 2)
+    trajectory[:, 0] = torch.arange(6) * math.cos(theta)
+    trajectory[:, 1] = torch.arange(6) * math.sin(theta)
+    return trajectory
+
+
+class Dataset(IterableDataset):
+    """
+    Implements an infinite dataset
+    """
+
+    def __iter__(self,):
+
+        while True:
+            image = torch.zeros((3, 100, 100))
+            agent_state_vector = torch.ones(3)
+            ground_truth = generate_trajectory(math.pi / 2)
+
+            yield image, agent_state_vector, ground_truth.unsqueeze(0)
+
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description='Run CoverNet to make sure it overfits on a single test case.')
+    parser.add_argument('--use_gpu', type=int, help='Whether to use gpu', default=0)
+    args = parser.parse_args()
+
+    if args.use_gpu:
+        device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
+    else:
+        device = torch.device('cpu')
+
+    dataset = Dataset()
+    dataloader = DataLoader(dataset, batch_size=16, num_workers=0)
+
+    backbone = MobileNetBackbone('mobilenet_v2')
+    model = CoverNet(backbone, num_modes=3, input_shape=(3, 100, 100))
+    model = model.to(device)
+
+    lattice = torch.zeros(3, 6, 2)
+    lattice[0] = generate_trajectory(math.pi / 2)
+    lattice[1] = generate_trajectory(math.pi / 4)
+    lattice[2] = generate_trajectory(3 * math.pi / 4)
+
+    loss_function = ConstantLatticeLoss(lattice)
+
+    current_loss = 10000
```

Comment:
This statement seems to have no effect (see line 88)?

Reply:
Got rid of it!